packages/psss/src/containers/UpcomingReportCard.js

```diff
@@ -0,0 +1,124 @@
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import Typography from '@material-ui/core/Typography';
+import { Warning } from '@material-ui/icons';
+import { CardContent, CardFooter, CardHeader, BarMeter, Card, Button } from '@tupaia/ui-components';
+import { openWeeklyReportsPanel, setActiveWeek } from '../store';
+import {
+  getWeekNumberByPeriod,
+  getFormattedStartByPeriod,
+  getFormattedEndByPeriod,
+} from '../utils';
+import { useUpcomingReport } from '../api/queries';
+import { REPORT_STATUSES, DUE_ISO_DAY } from '../constants';
+
+const StyledButton = styled(Button)`
+  margin-top: 1rem;
+  margin-bottom: 1rem;
+`;
+
+const DateSubtitle = styled(Typography)`
+  margin-top: 1rem;
+  margin-bottom: 1rem;
+  font-weight: 400;
+  color: ${props => props.theme.palette.text.secondary};
+`;
+
+const getDisplayDays = days => {
+  const d = Math.abs(days);
+  return `${Math.abs(d)} day${d > 1 ? 's' : ''}`;
+};
+
+const HeaderComponent = ({ status, days }) => {
+  const displayDays = getDisplayDays(days);
+
+  if (status === REPORT_STATUSES.UPCOMING && days <= 3) {
+    return (
+      <CardHeader
+        color="error"
+        title={`Confirmation due in ${displayDays}`}
+        label={<Warning color="error" />}
+      />
+    );
+  }
+
+  if (status === REPORT_STATUSES.UPCOMING && days === 0) {
+    return (
+      <CardHeader color="error" title="Confirmation due today" label={<Warning color="error" />} />
+    );
+  }
+
+  if (status === REPORT_STATUSES.OVERDUE) {
+    return (
+      <CardHeader
+        color="error"
+        title={`Confirmation overdue by ${displayDays}`}
+        label={<Warning color="error" />}
+      />
+    );
+  }
+
+  return <CardHeader title={`Confirmation due in ${displayDays}`} />;
+};
+
+HeaderComponent.propTypes = {
+  status: PropTypes.string.isRequired,
+  days: PropTypes.number.isRequired,
+};
+
+export const UpcomingReportCardComponent = ({ handleOpen }) => {
+  const { countryCode } = useParams();
+  const { isLoading, period, reportStatus, days } = useUpcomingReport(countryCode);
+
+  if (isLoading) {
+    return <Card variant="outlined" style={{ height: 300 }} />;
+  }
+
+  const buttonText =
+    reportStatus === REPORT_STATUSES.UPCOMING && days > DUE_ISO_DAY
+      ? 'View Now'
+      : 'Review and Confirm Now';
+
+  return (
+    <Card variant="outlined">
+      <HeaderComponent days={days} status={reportStatus} />
+      <CardContent>
+        <Typography variant="h4">Week {getWeekNumberByPeriod(period)}</Typography>
+        <Typography variant="h4" gutterBottom>
+          Upcoming Report
+        </Typography>
+        <DateSubtitle variant="subtitle2" gutterBottom>
+          {getFormattedStartByPeriod(period, 'LLL d, yyyy')} -{' '}
+          {getFormattedEndByPeriod(period, 'LLL d, yyyy')}
```

Comment:
I guess we could use [getDisplayDatesByPeriod](https://github.com/beyondessential/tupaia/blob/1628-handle-report-weeks/packages/psss/src/utils/period.js#L21) here? There's also another potential use case in [DateToolbar](https://github.com/beyondessential/tupaia/blob/1628-handle-report-weeks/packages/psss/src/components/DateToolbar/DateToolbar.js#L119).

Start date format is different in all those cases, but if consistency in the UI is beneficial in all those cases we should use the util

Reply:
hmmm I thought about this when putting it together and decided that they are different enough to be implemented separately. I think they are also the kind of thing a designer might tweak in one place or another but not everywhere etc

Reply:
In that case let's leave those as they are!